src/result-decorator.test.ts

```diff
@@ -23,60 +17,97 @@ describe('Result Decorator', () => {
             timestamp: '100',
         } as AxeResults;
 
-        const wcagInfo: DictionaryStringTo<WCAG[]> = {};
-        const resultDecorator = new ResultDecorator(wcagInfo);
+        const irrelevantWcagInfo: DictionaryStringTo<WCAGLinkData> = {};
+        const resultDecorator = new ResultDecorator(irrelevantWcagInfo);
+
         const decoratedResults: DecoratedAxeResults = resultDecorator.decorateResults(
-            resultStub,
+            inputWithNoIndividualResults,
         );
-        expect(decoratedResults.violations).toEqual([]);
-        expect(decoratedResults).toMatchSnapshot();
+
+        expect(decoratedResults).toMatchObject({
+            timestamp: '100',
+            targetPageUrl: 'https://example.com',
+            targetPageTitle: '',
+        });
     });
 
-    it('result decorator contains WCAG information that is provided as dependency', () => {
+    // tslint:disable-next-line: mocha-no-side-effect-code
+    const resultsArrayProperties = [
+        'passes',
+        'violations',
+        'inapplicable',
+        'incomplete',
+    ] as const;
+
+    // tslint:disable-next-line: mocha-no-side-effect-code
```

Comment:
I didn't want to sneak a tslint update into this same PR, but I'm on the fence about disabling this rule entirely; it's a valuable check (it prevents a class of bug we talked about in the last engineering discussion), but it's overzealous in that it also precludes the use `.each`, which isn't great.

Reply:
Peter suggested a way of doing it that was very reasonable/non-contentious, so I've included that.